Lib/fontTools/subset/__init__.py

```diff
@@ -2407,12 +2407,27 @@ def prune_post_subset(self, options):
             for subr in subrs.items:
                 subr.subset_subroutines (local_subrs, font.GlobalSubrs)
 
+        # Delete local SubrsIndex if empty
+        if hasattr(font, 'FDSelect'):
+            for fd in font.FDArray:
+                _delete_empty_subrs(fd)
+        else:
+            _delete_empty_subrs(font)
+
         # Cleanup
         for subrs in all_subrs:
             del subrs._used, subrs._old_bias, subrs._new_bias
 
     return True
 
+
+def _delete_empty_subrs(font):
```

Comment:
Maybe better to make this take a Private, not "font" which is sometimes a font, sometimes a fontDict.